src/Mod/Part/Gui/SectionCutting.cpp

```diff
@@ -0,0 +1,1475 @@
+                Refresh::XRange, Refresh::YRange, Refresh::ZRange);
+        }
+        else
+            refreshCutRanges(box);
+    }
+    // the case of an empty box and having cuts will be handles later by startCutting(true)
+
+    connect(ui->groupBoxX, &QGroupBox::toggled, this, &SectionCut::onGroupBoxXtoggled);
+    connect(ui->groupBoxY, &QGroupBox::toggled, this, &SectionCut::onGroupBoxYtoggled);
+    connect(ui->groupBoxZ, &QGroupBox::toggled, this, &SectionCut::onGroupBoxZtoggled);
+    connect(ui->cutX, QOverload<double>::of(&QDoubleSpinBox::valueChanged), this, &SectionCut::onCutXvalueChanged);
+    connect(ui->cutY, QOverload<double>::of(&QDoubleSpinBox::valueChanged), this, &SectionCut::onCutYvalueChanged);
+    connect(ui->cutZ, QOverload<double>::of(&QDoubleSpinBox::valueChanged), this, &SectionCut::onCutZvalueChanged);
+    connect(ui->cutXHS, &QSlider::sliderMoved, this, &SectionCut::onCutXHSsliderMoved);
+    connect(ui->cutYHS, &QSlider::sliderMoved, this, &SectionCut::onCutYHSsliderMoved);
+    connect(ui->cutZHS, &QSlider::sliderMoved, this, &SectionCut::onCutZHSsliderMoved);
+    connect(ui->flipX, &QPushButton::clicked, this, &SectionCut::onFlipXclicked);
+    connect(ui->flipY, &QPushButton::clicked, this, &SectionCut::onFlipYclicked);
+    connect(ui->flipZ, &QPushButton::clicked, this, &SectionCut::onFlipZclicked);
+    connect(ui->RefreshCutPB, &QPushButton::clicked, this, &SectionCut::onRefreshCutPBclicked);
+    
+    // if there is a cut, perform it
+    if (hasBoxX || hasBoxY || hasBoxZ) {
+        ui->RefreshCutPB->setEnabled(false);
+        startCutting(true);
+    }
+}
+
+// actions to be done when document was closed
+void SectionCut::noDocumentActions()
+{
+    blockSignals(true);
+    doc = nullptr;
+    // reset the cut group boxes
+    ui->groupBoxX->setChecked(false);
+    ui->groupBoxY->setChecked(false);
+    ui->groupBoxZ->setChecked(false);
+    ui->RefreshCutPB->setEnabled(true);
+    blockSignals(false);
+}
+
+void SectionCut::startCutting(bool isInitial)
+{
+    // there might be no document
+    if (!Gui::Application::Instance->activeDocument()) {
+        noDocumentActions();
+        return;
+    }
+    // the document might have been changed
+    if (doc != Gui::Application::Instance->activeDocument()->getDocument())
+        // refresh documents list
+        onRefreshCutPBclicked();
+
+    // we will reuse it several times
+    std::vector<App::DocumentObject*>::iterator it;
+
+    // delete the objects we might have already created to cut
+    // we must do this because we support several cuts at once and
+    // it is dangerous to deal with the fact that the user is free
+    // to uncheck cutting planes and to add/remove objects while this dialog is open
+    // We must remove in this order because the tree hierary of the features is Z->Y->X and Cut->Box
+    App::DocumentObject* anObject;
+    if (doc->getObject(CutZName)) {

```

Comment:
The code block below appears 6x inside the function (only the name is different). Write a lambda function to avoid code duplication.

Reply:
done in #6360